assets/js/rootLantiq.js

```diff
@@ -36,35 +36,34 @@ async function checkUbootUnlocked(serial) {
     return unlocked;
 }
 
-async function waitFailbackShell(writer, reader, outputMsgCallback) {
-    while (true) {
-        const { value, done } = await reader.read();
-
-        if (value.startsWith('Press the [f] key and hit [enter] to enter failsafe mode')) {
-            const interval = setInterval(function() {
-                writer.write('f\n');
-            }, 10);
-
-            outputMsgCallback("Root in progress: Trigger characters received. Waiting for boot to end...");
-            await delay(3000);
-            clearInterval(interval);
-            break;
+async function waitFailbackShell(serial, outputMsgCallback) {
+    await serial.readLine((line) => {
+        console.log(line);
+        if (line.startsWith('Press the [f] key and hit [enter] to enter failsafe mode')) {
+            return true;
         }
-    }
+    });
 
-    const interval = setInterval(function() {
-        writer.write(String.fromCharCode(10));
+    let interval = setInterval(function() {
+        serial.writeString('f\n');
```

Comment:
why? `f\n`?  `String.fromCharCode(10)` is only `\n`

Reply:
This is the previous interval that was moved out of the loop, it hasn't been changed just moved down